testlapack/dlaqp2.go

```diff
@@ -0,0 +1,60 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"testing"
+)
+
+type Dlaqp2er interface {
+	Dlaqp2(m, n, offset int, a []float64, lda int, jpvt []int, tau, vn1, vn2, work []float64)
+}
+
+// TODO(kortschak) Make this a test of more than just indexing failure and "the vibe".
```

Comment:
You could check that `A[offset:m, 0:n]` contains its QR decomposition. Similarly for Dlaqps but only for the block `A[offset:m, 0:kb]`.  It's not so much work, just copy, paste and adjust the test for Dgeqp3. It's up to you.